src/main/java/jenkins/advancedqueue/jobinclusion/strategy/FolderBasedJobInclusionStrategy.java

```diff
@@ -62,19 +66,70 @@ public ListBoxModel getListFolderItems() {
 
 	};
 
+	static public class JobPattern {
```

Comment:
> I've been looking at how it's implemented in ViewBasedJobInclusionStrategy and it doesn't seem to be required there.

Well, I was not an author of that :( I rather suggest a proper design from the Jenkins standpoint. The current implementation is rather a hack based on the JSON structure presumptions. It should work, but I am not sure if we can guarantee compatibility of it in the Jenkins core

If you want to keep the class as is, mark it as `@Restricted(NoExternalUse.class)` to prevent the external usage. But it will also require the constructor modification then.